src/main/java/net/geforcemods/securitycraft/blocks/CageTrapBlock.java

```diff
@@ -47,12 +51,30 @@ public CageTrapBlock(Material material) {
 
 	@Override
 	public VoxelShape getCollisionShape(BlockState state, IBlockReader world, BlockPos pos, ISelectionContext ctx){
-		if(ctx instanceof EntitySelectionContext && ((EntitySelectionContext)ctx).getEntity() instanceof PlayerEntity)
-		{
+		if(ctx instanceof EntitySelectionContext) {
 			TileEntity te = world.getTileEntity(pos);
 
-			if(te instanceof IOwnable && ((IOwnable)te).getOwner().isOwner((PlayerEntity)((EntitySelectionContext)ctx).getEntity()))
-				return VoxelShapes.fullCube();
+
+
+			if (((EntitySelectionContext) ctx).getEntity() instanceof PlayerEntity) {
+				if (te instanceof IOwnable && ((IOwnable) te).getOwner().isOwner((PlayerEntity) ((EntitySelectionContext) ctx).getEntity()))
+					return super.getCollisionShape(state, world, pos, ctx);
```

Comment:
This is incorrect. You need to return VoxelShapes.fullCube() when the Cage Trap is not disguised, otherwise return super.